hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestInsertTable.scala

```diff
@@ -2160,6 +2172,7 @@ class TestInsertTable extends HoodieSparkSqlTestBase {
            |union
            |select '1' as id, 'aa' as name, 123 as dt, '2023-10-12' as `day`, 12 as `hour`
            |""".stripMargin)
+      spark.sparkContext.addSparkListener(new StageParallelismListener(stageName = "collect at HoodieSparkEngineContext.java"))
```

Comment:
@xuzifu666 : Relying of class name string here would be highly fragile here. Can you make this explicit dependent to the class and also validate if the callback is indeed being called.

Reply:
Thanks，good advice. I would change it @bvaradar

Reply:
Hi，I had try dependent to the class，in this case all query stage would relate to HoodieSparkEngineContext class，so change it with this class to check @bvaradar PTAL